runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/commits/StreamingEngineWorkCommitter.java

```diff
@@ -61,11 +62,10 @@ public final class StreamingEngineWorkCommitter implements WorkCommitter {
       Supplier<CloseableStream<CommitWorkStream>> commitWorkStreamFactory,
       int numCommitSenders,
       Consumer<CompleteCommit> onCommitComplete,
-      String backendWorkerToken) {
+      String backendWorkerToken,
+      WeightedSemaphore<Commit> weigher) {
```

Comment:
rename weigher throughout to commitByteSemaphore or something capturing it is the limiter not just a weighing function?

Reply:
done